tests/client/core/i18n/test_utils.js

```diff
@@ -445,9 +445,34 @@ describe('i18n utils', () => {
       expect(i18n.formatNumber(9518231)).toEqual('9,518,231');
     });
 
-    it('localised formatting a number', () => {
+    it('provides a number localized by locale', () => {
+      const numberFormatSpy = sinon.spy(Intl, 'NumberFormat');
       const i18n = utils.makeI18n({}, 'de', FakeJed);
       expect(i18n.formatNumber(9518231)).toEqual('9.518.231');
+      sinon.assert.calledWith(numberFormatSpy, 'de');
+    });
+
+    it('calls Intl.NumberFormat.format', () => {
+      const formatStub = sinon.stub();
+      const fakeIntl = {
+        NumberFormat: () => {
+          return {
+            format: formatStub,
+          };
+        },
+      };
+      const i18n = utils.makeI18n({}, 'de', FakeJed, { _Intl: fakeIntl });
+      i18n.formatNumber(9518231);
+      sinon.assert.calledWith(formatStub, 9518231);
+    });
+
+    it('falls-back to number.toLocaleString if Intl is undefined', () => {
+      const i18n = utils.makeI18n({}, 'de', FakeJed, { _Intl: false });
+      const numberFormatSpy = sinon.spy(Intl, 'NumberFormat');
+      expect(i18n.formatNumber(1234)).toEqual('1,234');
+      // The calling of toLocalString is inferred by not calling
+      // Intl.formatNumber since there's no way to mock a prototype builtin.
+      sinon.assert.notCalled(numberFormatSpy);
     });
   });
```

Comment:
I think it would be helpful to add another test that simulates the case where `Intl.NumberFormat` is undefined, like  `utils.makeI18n(..., { _Intl: {} })`